oak/server/rust/oak_runtime/src/node/logger.rs

```diff
@@ -36,12 +38,24 @@ fn logger(pretty_name: &str, runtime: RuntimeRef, reader: ChannelReader) -> Resu
         let _ = runtime.wait_on_channels(&[Some(&reader)]);
 
         if let Some(message) = runtime.channel_read(&reader)? {
-            let message = String::from_utf8_lossy(&message.data);
-            info!("{} LOG: {}", pretty_name, message);
+            if let Ok(msg) = LogMessage::decode(&*message.data) {
+                info!(
+                    "{} LOG: {} {}:{}: {}",
+                    pretty_name,
+                    to_level_name(msg.level),
+                    msg.file,
+                    msg.line,
+                    msg.message
+                )
+            };
```

Comment:
Is the semicolon needed?  Also, maybe an error message if the `decode` fails?

Reply:
Thanks. Removed the semicolon!
I was also wondering if an error message is needed. The comment above this chunk of code says that an error indicates the termination of the runtime, and therefore is ignored. `decode` may fail for other reasons as well, I guess. I have added an error message.

Reply:
There's a few things going on here, and I think the error behaviour for each of them is a bit different.
 - First, there's a `wait_on_channels`.  This should always succeed, unless the runtime is terminating.
 - Next is the `channel_read`.  This will fail if the channel is closed (or if the runtime is terminating), and then the `?` will lead to the loop exiting and the pseudo-Node shutting down.
 - Finally is the `decode`.  This will fail if the (third-party) Node code has sent us garbage bytes; the loop can carry on in the (probably vain) hope that the next message to arrive is well-formed.